src/shims/unix/linux_like/syscall.rs

```diff
@@ -15,8 +15,8 @@ pub fn syscall<'tcx>(
     dest: &MPlaceTy<'tcx>,
 ) -> InterpResult<'tcx> {
     // We do not use `check_shim` here because `syscall` is variadic. The argument
-    // count is checked bellow.
-    ecx.check_abi_and_shim_symbol_clash(abi, Conv::C, link_name)?;
+    // count is checked below.
+    let ([op], _var) = ecx.check_shim_variadic::<1>(abi, Conv::C, link_name, "syscall", args)?;
```

Comment:
```suggestion
    let ([op], _var) = ecx.check_shim_variadic(abi, Conv::C, link_name, "syscall", args)?;
```

is that explicit length necessary? doesn't it get inferred from the return type's array length?

Reply:
Oh yea, it does